building/build-debs/homeworld-admin-tools/src/configuration.py

```diff
@@ -164,6 +164,27 @@ def get_node(self, node_name: str) -> Node:
                 return node
         command.fail("no such node: %s" % node_name)
 
+    def has_hostname(self, hostname: str) -> bool:
+        if hostname.endswith(self.external_domain):
+            # strip external domain
+            hostname = hostname[:-(len(self.external_domain) + 1)]
+        elif hostname.endswith(self.internal_domain):
+            # strip internal domain
+            hostname = hostname[:-(len(self.internal_domain) + 1)]
+        return self.has_node(hostname)
+
+    def get_hostname(self, name: str) -> str:
```

Comment:
Nitpick: it might be worth making the point of this function clearer in the name. Maybe `lookup_hostname` or `validate_domain` or `convert_fqdn` or something along one of those lines?